applications/crossbar/src/modules/cb_migrations.erl

```diff
@@ -0,0 +1,273 @@
+                     kz_json:set_value(I, D, Acc)
+             end,
+
+    Resp     = lists:foldl(Format, kz_json:new(), ?MIGRATIONS_LIST),
+    Context1 = cb_context:set_resp_data(Context, Resp),
+
+    cb_context:set_resp_status(Context1, 'success').
+
+-spec load_migration_summary(binary(), cb_context:context()) -> cb_context:context().
+load_migration_summary(MigId, Context) ->
+    case lists:keyfind(MigId, 1, ?MIGRATIONS_LIST) of
+        'false' ->
+            Context1 = cb_context:set_resp_data(Context, <<"migration not found">>),
+            cb_context:set_resp_error_code(Context1, 404);
+
+        Migration ->
+            render_migration_summary(Migration, Context)
+    end.
+
+-spec render_migration_summary({binary(), binary(), atom()}, cb_context:context()) -> cb_context:context().
+render_migration_summary({Id, Desc, _Callback}, Context) ->
+    Base = [
+            {<<"id">>, Id}
+           ,{<<"description">>, Desc}
+           ],
+
+    Perf = get_migration_performed(Id, Context),
+    Resp = kz_json:from_list(Base ++ Perf),
+
+    Context1 = cb_context:set_resp_data(Context, Resp),
+    cb_context:set_resp_status(Context1, 'success').
+
+-spec get_migration_performed(binary(), cb_context:context()) -> list().
+get_migration_performed(Id, Context) ->
+    case kz_datamgr:open_cache_doc(cb_context:account_db(Context), ?MIGRATIONS_DOC) of
+        {'ok', Doc} ->
+            check_migration_performed(Id, Doc);
+
+        _Other ->
+            [{<<"performed">>, 'false'}]
+    end.
+
+-spec check_migration_performed(binary(), kz_json:object()) -> list().
+check_migration_performed(Id, Doc) ->
+    case kz_json:get_value([<<"migrations_performed">>, Id], Doc) of
+        'undefined' ->
+            [{<<"performed">>, 'false'}];
+
+        Value ->
+            [{<<"performed">>, 'true'}, {<<"performed_time">>, kz_json:get_value(<<"performed_time">>, Value)}]
+    end.
+
+-spec maybe_perform_migration(binary(), cb_context:context()) -> cb_context:context().
+maybe_perform_migration(MigId, Context) ->
+    case kz_json:get_value(<<"perform_migration">>, cb_context:req_data(Context)) of
+        <<"now">> ->
+            check_migration_valid(MigId, Context);
+
+        _Other ->
+            Resp = kz_json:from_list([
+                                      {<<"error">>, <<"invalid action time">>}
+                                     ]),
+            Context1 = cb_context:set_resp_data(Context, Resp),
+            cb_context:set_resp_status(Context1, 'failed')
+    end.
+
+-spec check_migration_valid(binary(), cb_context:context()) -> cb_context:context().
+check_migration_valid(MigId, Context) ->
+    case lists:keyfind(MigId, 1, ?MIGRATIONS_LIST) of
+        'false' ->
+            Resp = kz_json:from_list([
+                                      {<<"error">>, <<"invalid migration">>}
+                                     ]),
+            Context1 = cb_context:set_resp_data(Context, Resp),
```

Comment:
@mark2600 You should use cb_context:add_validation_error so that the payloads match the standard errors